code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -0,0 +1,821 @@
+		message_admins("<span class='notice'>[key_name_admin(src)] made [key_name_admin(current_robot)] say: [message]</span>")
+		// don't sanitize again
+		current_robot.say(message, null, FALSE, ignore_speech_problems, ignore_atmospherics, ignore_languages)
+		return TRUE
+
+	var/message_mode = parse_message_mode(message, "headset")
+
+	if(message_mode)
+		if(message_mode == "headset")
+			message = copytext(message, 2)
+		else
+			message = copytext(message, 3)
+
+	message = trim_left(message)
+
+	var/list/message_pieces = list()
+	if(ignore_languages)
+		message_pieces = message_to_multilingual(message)
+	else
+		message_pieces = parse_languages(message)
+
+	// hivemind languages
+	if(istype(message_pieces, /datum/multilingual_say_piece))
+		var/datum/multilingual_say_piece/S = message_pieces
+		S.speaking.broadcast(src, S.message)
+		return TRUE
+
+	if(!LAZYLEN(message_pieces))
+		. = FALSE
+		CRASH("Message failed to generate pieces. [message] - [json_encode(message_pieces)]")
+
+	create_log(SAY_LOG, "[message_mode ? "([message_mode])" : ""] '[message]'")
+
+	playsound(get_turf(src), pick(speech_sounds), 30, TRUE)
+	if(istype(loc, /obj/item/radio))
+		var/obj/item/radio/R = loc
+		name = gen_speech_name()
+		R.talk_into(src, message_pieces, message_mode, verbage = verb)
+		name = real_name
+		return TRUE
+	else if(istype(loc, /obj/machinery/hologram/holopad))
+		var/obj/machinery/hologram/holopad/H = loc
+		name = "[H]"
+		for(var/mob/M in get_mobs_in_view(7, H))
+			M.hear_say(message_pieces, verb, FALSE, src)
+		name = real_name
+		return TRUE
+
+	emote("me", message = "[pick(emote_hear)]")
+	return TRUE
+
+/mob/living/simple_animal/demon/pulse_demon/update_runechat_msg_location()
+	if(istype(loc, /obj/machinery/hologram/holopad))
+		runechat_msg_location = loc.UID()
+	else
+		return ..()
+
+/mob/living/simple_animal/demon/pulse_demon/visible_message(message, self_message, blind_message)
+	// overriden because pulse demon is quite often in non-turf locs, and /mob/visible_message acts differently there
+	for(var/mob/M in get_mobs_in_view(7, src))
+		if(M.see_invisible < invisibility)
+			continue //can't view the invisible
+		var/msg = message
+		if(self_message && M == src)
+			msg = self_message
+		M.show_message(msg, EMOTE_VISIBLE, blind_message, EMOTE_AUDIBLE)
+
+/mob/living/simple_animal/demon/pulse_demon/has_internal_radio_channel_access(mob/user, list/req_one_accesses)
+	return has_access(list(), req_one_accesses, get_all_accesses())
+
+/mob/living/simple_animal/demon/pulse_demon/proc/try_hijack_apc(obj/machinery/power/apc/A, remote = FALSE)
+	// one APC per pulse demon, one pulse demon per APC, no duplicate APCs
+	if(!is_valid_apc(A) || (A in hijacked_apcs) || apc_being_hijacked || A.being_hijacked)
+		return FALSE
+
+	to_chat(src, "<span class='notice'>You are now attempting to hijack [A], this will take approximately [hijack_time / 10] seconds.</span>")
```

Comment:
```suggestion
	to_chat(src, "<span class='notice'>You are now attempting to hijack [A], this will take approximately [hijack_time] seconds.</span>")
```
Hijack time is already defined in seconds

Reply:
SECONDS just multiplies stuff by 10, it's still 300 internally